src/node_platform.h

```diff
@@ -22,16 +22,69 @@ class PerIsolatePlatformData;
 template <class T>
 class TaskQueue {
  public:
+  class Locked {
+   public:
+    explicit Locked(TaskQueue* queue) : queue_(queue), lock_(queue->lock_) {}
+
+    void Push(std::unique_ptr<T> task) {
+      queue_->outstanding_tasks_++;
+      queue_->task_queue_.push(std::move(task));
+      queue_->tasks_available_.Signal(lock_);
+    }
```

Comment:
The convention in node is to move method bodies into either the corresponding .cc or a -inl.h file (with the exception of really simple getters and setters, the kind you write on a single line.)